File: AtCoder/abs/haskell/abc086C.hs
import Control.Monad
check [t1,x1,y1] [t2,x2,y2] =
    (dt >= dx + dy) && ((dt-dx-dy) `mod` 2 == 0)
    where dx = abs(x2-x1)
          dy = abs(y2-y1)
          dt = t2-t1

ok [a,b] = check a b
ok (a:b:xs) = check a b && ok (b:xs)

main = do
    n <- readLn
    txys <- replicateM n $ map read . words <$> getLine
    putStrLn $ if ok ([0,0,0]:txys) then "Yes" else "No"

File: AtCoder/abs/ocaml/abc086C.ml
open Batteries

let n = read_int()

let rec check (t1,x1,y1) (t2,x2,y2) =
    let dx, dy, dt = abs(x2-x1), abs(y2-y1), t2-t1 in
    (dt >= dx + dy) && ((dt-dx-dy) mod 2 = 0)

let rec ok = function
    | p1::p2::ps -> check p1 p2 && ok (p2::ps)
    | _ -> true

let ts = List.init n (fun _ -> Scanf.sscanf (read_line ()) "%d %d %d" (fun t x y -> t, x, y))
let () = (if ok ((0,0,0)::ts) then "Yes" else "No") |> print_endline